src/module/pageSettings.ts

```diff
@@ -31,12 +31,12 @@ export const pageSettingsSlice = createSlice({
     ) {
       return { ...state, ...defaultPageSettings, ...action.payload };
     },
-    setPageParams(state, action: PayloadAction<Record<string, string>>) {
-      state.searchParams = action.payload;
+    setPageParams(state, action: PayloadAction<URLSearchParams>) {
+      state.searchParams = action.payload as unknown as Map<string, string | null>; // Ok, did not get why it does not like the Map<string, string | null> type
       return state;
     },
```

Comment:
```suggestion
    setPageParams: (state, action: PayloadAction<URLSearchParams>) => ({
      ...state,
      searchParams: action.payload,
    }),
```
On peut imaginer aussi faire pareil pour le reducer juste au dessus :
```tsx
    setPageSettings: (
      state,
      action: PayloadAction<Pick<PageSettingsSlice, InternalPageSettingsKeys> & Partial<PageSettings>>,
    ) => ({ ...state, ...defaultPageSettings, ...action.payload }),
```

Reply:
mouais, pour le coup je préfère quand même la notation "fonctionnelle" plutôt que "lambdaïque", ça refait penser à la lisibilité de la POO à mon avis